rust/src/cliwrap/kernel_install.rs

```diff
@@ -0,0 +1,79 @@
+// If not running on container continue the current path.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+use crate::cliwrap::cliutil;
+use anyhow::{anyhow, Context, Result};
+use camino::Utf8Path;
+use cap_std::fs::FileType;
+use cap_std::fs_utf8::Dir as Utf8Dir;
+use cap_std_ext::cap_std;
+use std::process::Command;
+
+/// Primary entrypoint to running our wrapped `kernel-install` handling.
+pub(crate) fn main(argv: &[&str]) -> Result<()> {
+    if !ostree_ext::container_utils::is_ostree_container()? {
+        return cliutil::exec_real_binary("kernel-install", argv);
+    }
+    let is_install = matches!(argv.get(0), Some(&"add"));
+
+    let modules_path = Utf8Dir::open_ambient_dir("lib/modules", cap_std::ambient_authority())?;
+    //kernel-install is called by kernel-core and kernel-modules cleanup let's make sure we just call dracut once.
+    let mut new_kernel: Option<_> = None;
+
+    //finds which the kernel to remove and which to generate the initramfs for.
+    for entry in modules_path.entries()? {
+        let entry = entry?;
+        let kernel_dir = entry.file_name()?;
+        let kernel_binary_path = Utf8Path::new(&kernel_dir).join("vmlinuz");
+
+        if entry.file_type()? == FileType::dir() {
+            if modules_path.exists(kernel_binary_path) {
+                if is_install {
+                    new_kernel = Some(kernel_dir);
+                }
+            } else {
+                new_kernel = None;
+                modules_path.remove_dir_all(kernel_dir)?;
+            }
+        }
+    }
+    if let Some(k) = new_kernel {
+        run_dracut(&k)?;
+    }
+    Ok(())
+}
+
+fn run_dracut(kernel_dir: &str) -> Result<()> {
+    //SET Environment variable needed for Dracut
+    std::env::set_var("BYPASS_DRACUT_WRAP", "TRUE");
```

Comment:
One tricky thing here is that setting global environment variables is not threadsafe; 
https://internals.rust-lang.org/t/synchronized-ffi-access-to-posix-environment-variable-functions/15475

We can handle this by calling `Command::set_env()` which will only do it for the child and not mutate *our* environment.

Reply:
Also if we do go envvar, let's namespace it like `RPMOSTREE_CLIWRAP_SKIP` or something?

Reply:
We discussed the env bits in a live chat, I think this env var is OK as is but we could have a `RPMOSTREE_CLIWRAP_SKIP=1` that applies to *all* cliwrap code; should be a pretty easy followup.

Reply:
removed the env var here in favor of the global one we discussed.